staging/src/k8s.io/legacy-cloud-providers/vsphere/vclib/connection.go

```diff
@@ -217,6 +217,13 @@ func (connection *VSphereConnection) NewClient(ctx context.Context) (*vim25.Clie
 		connection.RoundTripperCount = RoundTripperDefaultCount
 	}
 	client.RoundTripper = vim25.Retry(client.RoundTripper, vim25.TemporaryNetworkError(int(connection.RoundTripperCount)))
+	vcdeprecated, err := isvCenterDeprecated(client.ServiceContent.About.Version, client.ServiceContent.About.ApiVersion)
+	if err != nil {
+		klog.Errorf("failed to check if vCenter version:%v and api version: %s is deprecated. Error: %v", client.ServiceContent.About.Version, client.ServiceContent.About.ApiVersion, err)
+	}
+	if vcdeprecated {
+		klog.Warningf("vCenter is deprecated. version: %s, api verson: %s Please consider upgrading vCenter and ESXi servers to 6.7u3 or higher", client.ServiceContent.About.Version, client.ServiceContent.About.ApiVersion)
```

Comment:
Current logic doesn't check ESXi versions. Is it possible to do those checks from here? If so, can we check and print the ESXi warning accordingly?

Reply:
We do not connect to the ESXi hosts so we can not perform ESXi version check.

Reply:
I don't know the details, but what does mean by "not connect to ESXi hosts"? Can vCenter client list all hosts?

Reply:
We can, but we do not have any workflow where we need to list all ESXi hosts and retrieve their versions.